src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/ListenerContext.cs

```diff
@@ -28,18 +28,65 @@ protected UvStreamHandle CreateAcceptSocket()
             switch (EndPointInformation.Type)
             {
                 case ListenType.IPEndPoint:
-                case ListenType.FileHandle:
-                    var tcpHandle = new UvTcpHandle(TransportContext.Log);
-                    tcpHandle.Init(Thread.Loop, Thread.QueueCloseHandle);
-                    tcpHandle.NoDelay(EndPointInformation.NoDelay);
-                    return tcpHandle;
+                    return AcceptTcp();
                 case ListenType.SocketPath:
-                    var pipeHandle = new UvPipeHandle(TransportContext.Log);
-                    pipeHandle.Init(Thread.Loop, Thread.QueueCloseHandle);
-                    return pipeHandle;
+                    return AcceptPipe();
+                case ListenType.FileHandle:
+                    return AcceptHandle();
                 default:
                     throw new InvalidOperationException();
             }
         }
+
+        private UvTcpHandle AcceptTcp()
+        {
+            var socket = new UvTcpHandle(TransportContext.Log);
+
+            try
+            {
+                socket.Init(Thread.Loop, Thread.QueueCloseHandle);
+                socket.NoDelay(EndPointInformation.NoDelay);
+            }
+            catch
+            {
+                socket.Dispose();
```

Comment:
👍  Good catch. Better to explicitly dispose than rely on the finalizer when one of these initialization methods throws.